--- calendar.l	2024-01-01 21:04:34.491559560 +0100
+++ calendar.lisp	2024-01-01 21:01:45.656997478 +0100
@@ -1,3 +1,8 @@
+;;;; -*- mode: Lisp; indent-tabs-mode: nil; -*-
+;;;; This file is an automatically generated and modified version of 
+;;;; calendar.l. The original may be found here: 
+;;;; https://www.cs.tau.ac.il/~nachum/calendar-book/papers/calendar.l
+
 ;; The following Lisp code is from ``Calendrical
 ;; Calculations'' by Nachum Dershowitz and Edward
 ;; M. Reingold, Software---Practice & Experience, vol. 20,
@@ -9,7 +14,6 @@
 
 ;; This code is in the public domain, but any use of it
 ;; should publically acknowledge its source.
-
 (defun quotient (m n)
    (floor (/ m n)))
 
@@ -228,17 +232,6 @@
 ;; Else return
     12))
 
-(defun last-day-of-hebrew-month (month year)
-;; Last day of $month$ in Hebrew $year$.
-  (if (or (member month (list 2 4 6 10 13))
-          (and (= month 12) (not (hebrew-leap-year year)))
-          (and (= month 8) (not (long-heshvan year)))
-          (and (= month 9) (short-kislev year)))
-;; Then return
-      29
-;; Else return
-    30))
-
 (defun hebrew-calendar-elapsed-days (year)
 ;; Number of days elapsed from the Sunday prior to the start of the
 ;; Hebrew calendar to the mean conjunction of Tishri of Hebrew $year$.
@@ -314,6 +307,17 @@
 ;; True if Kislev is short in Hebrew $year$.
   (= (mod (days-in-hebrew-year year) 10) 3))
 
+(defun last-day-of-hebrew-month (month year)
+  ;; Last day of $month$ in Hebrew $year$.
+  (if (or (member month (list 2 4 6 10 13))
+          (and (= month 12) (not (hebrew-leap-year year)))
+          (and (= month 8) (not (long-heshvan year)))
+          (and (= month 9) (short-kislev year)))
+      ;; Then return
+      29
+      ;; Else return
+      30))
+
 (defun absolute-from-hebrew (date)
 ;; Absolute date of Hebrew $date$.
   (let* ((month (extract-month date))
@@ -587,10 +591,10 @@
      (t (absolute-from-hebrew
          (list death-month death-day year))))))
 
-(defconstant mayan-days-before-absolute-zero
+(defparameter mayan-days-before-absolute-zero
 ;; Number of days of the Mayan calendar epoch before absolute day 0,
 ;; according to the Goodman-Martinez-Thompson correlation.
-  1137140)
+  1137142) ; changed as in Reingold/Dershowitz 2018; Original value: 1137140
 
 (defun absolute-from-mayan-long-count (count)
 ;; Absolute date corresponding to the Mayan long count $count$, 
@@ -616,10 +620,7 @@
          (kin (mod day-of-tun 20)))
     (list baktun katun tun uinal kin)))
 
-(defun quotient (m n)
-   (floor m n))
-
-(defconstant mayan-haab-at-epoch '(8 18))
+(defparameter mayan-haab-at-epoch '(8 18))
 
 (defun mayan-haab-from-absolute (date)
 ;; Mayan haab date of absolute date $date$.
@@ -653,7 +654,7 @@
 ;; Positive remainder of $m/n$ with $n$ instead of 0.
   (1+ (mod (1- m) n)))
 
-(defconstant mayan-tzolkin-at-epoch '(4 20))
+(defparameter mayan-tzolkin-at-epoch '(4 20))
 
 (defun mayan-tzolkin-from-absolute (date)
 ;; Mayan tzolkin date of absolute date $date$.
@@ -705,14 +706,6 @@
                 18980))
       nil)));;  haab-tzolkin combination is impossible.
 
-(defun french-last-day-of-month (month year)
-;; Last day of {\em month, year} on the French Revolutionary calendar.
-  (if (< month 13)
-      30
-    (if (french-leap-year year)
-        6
-      5)))
-
 (defun french-leap-year (year)
 ;; True if {\em year} is a leap year on the French Revolutionary calendar.
   (or (member year '(3 7 11));; Actual.
@@ -722,6 +715,14 @@
            (not (member (mod year 400) '(100 200 300)))
            (not (= 0 (mod year 4000))))))
 
+(defun french-last-day-of-month (month year)
+  ;; Last day of {\em month, year} on the French Revolutionary calendar.
+  (if (< month 13)
+      30
+      (if (french-leap-year year)
+          6
+          5)))
+
 (defun absolute-from-french (date)
 ;; Absolute date of French Revolutionary {\em date}.
   (let* ((month (first date))
@@ -765,10 +766,10 @@
                (1- (absolute-from-french (list month 1 year))))))
     (list month day year))))
 
-(defconstant solar-sidereal-year (+ 365 279457/1080000))
-(defconstant solar-month (/ solar-sidereal-year 12))
-(defconstant lunar-sidereal-month (+ 27 4644439/14438334))
-(defconstant lunar-synodic-month (+ 29 7087771/13358334))
+(defparameter solar-sidereal-year (+ 365 279457/1080000))
+(defparameter solar-month (/ solar-sidereal-year 12))
+(defparameter lunar-sidereal-month (+ 27 4644439/14438334))
+(defparameter lunar-synodic-month (+ 29 7087771/13358334))
 
 (defun solar-longitude (days) 
 ;; Mean sidereal longitude of the sun, in degrees,
